public/app/plugins/panel/geomap/editor/StyleRuleEditor.tsx

```diff
@@ -40,7 +40,21 @@ export const StyleRuleEditor: FC<StandardEditorProps<FeatureStyleConfig, any, an
   const uniqueSelectables = useMemo(() => {
     const key = value?.check?.property;
     if (key && feats && value.check?.operation === ComparisonOperation.EQ) {
-      return getUniqueFeatureValues(feats, key).map((v) => ({ value: v, label: v }));
+      return getUniqueFeatureValues(feats, key).map((v) => {
```

Comment:
hymm -- rather than adding this funky business, maybe we should check when calling `onChangeValue`?

The weird part is that the number vs string distinction should really come from the feature value, not the comparison value :thinking:

Reply:
I originally had these checks in the `onChangeValue` but that resulted in the value select to have `(not found)` appended to each value due to number and string inequality. It seems that we want to store number values as numbers for best UX (hence during initialization step vs. later on)